app/src/main/java/com/allan/boardbuddies/NoteAdapter.java

```diff
@@ -9,20 +9,24 @@
 import android.view.LayoutInflater;
 import android.widget.TextView;
 
+import com.allan.boardbuddies.fragments.NotesFragment;
+
 // binds data to views displayed by RecyclerView
 public class NoteAdapter extends RecyclerView.Adapter<NoteAdapter.NoteViewHolder> {
     private ArrayList<Note> notes;
+    private OnNoteListener onNoteListener;
     //constructor
-    public NoteAdapter(ArrayList<Note> notes){
+    public NoteAdapter(ArrayList<Note> notes, OnNoteListener onNoteListener){
         this.notes = notes;
+        this.onNoteListener = onNoteListener;
     }
 
     //Creates ViewHolder for view (note_item.xml), converts xml to View object, this method runs when no ViewHolders are available
     @NonNull
     @Override
     public NoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
         View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.note_item, parent, false);
-        return new NoteViewHolder(itemView);
+        return new NoteViewHolder(itemView, this.onNoteListener);
```

Comment:
I think what you want to do, instead of passing this listener into the view holder, is call the overridable method `onBindViewHolder(view holder, position)` from within the adapter. Here, we can set the OnClickListener directly because we have the notes and the position.

The adapter is in charge of binding the data, including on click actions, to the view holder. It's basically the bridge between the view and the UI.

https://developer.android.com/develop/ui/views/layout/recyclerview#implement-adapter

Reply:
![image](https://github.com/allanfang1/boardbuddies/assets/32990185/9ec81df8-5061-4595-b3bf-1d272fd78e16)

https://stackoverflow.com/questions/31627073/why-does-onclicklistener-on-a-viewholder-dont-work

I think it's not recommended?

Reply:
Also I would need to pass the context to the adaptor for me to create new activity (intent) which may be extra...

Reply:
But I guess I could do something like this: 
https://stackoverflow.com/questions/49969278/recyclerview-item-click-listener-the-right-way
Thoughts?

Reply:
Ok, you're right. Let's keep it this way. One more comment for the click listener below though